### PR TITLE
feat(lasso): extend LassoGuardrail to support tool calling (RND-5748)

### PR DESCRIPTION
**Pull request by vladpolevoi**
[RND-5748](https://linear.app/lasso-security/issue/RND-5748/tool-calling-litellm-extend-lassoguardrail-python-class-to-support)

## Summary
- `_prepare_payload` now maps `data["tools"]` (OpenAI `ChatCompletionToolParam`) to Lasso `ToolDefinition` shape and includes it in the request body
- New `_expand_messages_for_classification` helper converts OpenAI-format messages to Lasso content blocks: assistant `tool_calls` → `tool_use` blocks, `role=tool` messages → `developer` role + `tool_result` blocks
- `async_post_call_success_hook` now extracts `choice.message.tool_calls` from the LLM response and classifies them as `tool_use` blocks
- Existing text-only behavior unchanged

## Test plan
- [ ] `test_payload_preparation_with_tools` — tools field mapped and included
- [ ] `test_payload_preparation_no_tools` — no tools key when not provided (regression)
- [ ] `test_expand_messages_assistant_tool_calls` — tool_calls expand to tool_use blocks
- [ ] `test_expand_messages_tool_role` — role=tool folds into tool_result block
- [ ] `test_expand_messages_tool_role_missing_tool_call_id` — malformed tool message skipped
- [ ] `test_expand_messages_assistant_with_text_and_tool_calls` — text + tool_calls produces two messages
- [ ] `test_expand_messages_plain_text_unchanged` — plain text regression
- [ ] `test_post_call_with_tool_calls` — tool_calls in response classified correctly
- [ ] `test_post_call_text_only_regression` — text-only post-call still works

## Summary by CodeRabbit

* **New Features**
  * Lasso guardrail now supports OpenAI-style tool/function calls end-to-end, converting calls and arguments into structured blocks and including tool definitions in payloads.

* **Bug Fixes**
  * Masking preserves and correctly maps masked assistant text and tool-call arguments back to the original message schema so model responses apply masks accurately.

* **Tests**
  * Added comprehensive tests for tool-calling payloads, message expansion, mixed text+tool responses, masking round-trips, and related regressions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/lasso-security/litellm/pull/54)

### CODE REVIEW

File: litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py [hunk 1]
```diff
@@ -478,6 +622,97 @@ def _log_masking_applied(
...
+                            parsed = json.loads(args_str)
+                            if isinstance(parsed, dict):
+                                input_data = parsed
+                        except (json.JSONDecodeError, TypeError):
+                            verbose_proxy_logger.warning(
+                                "Failed to parse tool_call arguments, dropping input"
+                            )
+                    expanded.append(
+                        {
+                            "role": "assistant",
```
Comment (orgersh92) [inline litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py]: In Lasso we have `"model"` role, which is I think better fit for tool_use request.

When LLM is returning text contant, we mark this event as "assistant" since it should "meet the user".
But otherwise, if it is only "sending instructions  to the developer" - then "model" is a better role

File: litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py [hunk 2]
```diff
@@ -219,11 +220,58 @@ async def async_post_call_success_hook(
...
+                            parsed = json.loads(args_str)
+                            if isinstance(parsed, dict):
+                                input_data = parsed
+                        except (json.JSONDecodeError, TypeError):
+                            verbose_proxy_logger.debug(
+                                "Failed to parse tool_call arguments as JSON, sending null input"
+                            )
+                    response_messages.append(
+                        {
+                            "role": "assistant",
```
Comment (orgersh92) [inline litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py]: same comments about (`"model"` vs. `"assistant"`)

File: litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py [hunk 3]
```diff
@@ -412,16 +463,109 @@ async def _handle_masking(
...
+                    data["messages"], response["messages"]
+                )
                 self._log_masking_applied(message_type, dict(response))
 
             return data
         except Exception as e:
             await self._handle_api_error(e, message_type)
             return data  # This line won't be reached due to exception, but satisfies type checker
 
+    def _map_masked_messages_back(
```
Comment (orgersh92) [inline litellm/proxy/guardrails/guardrail_hooks/lasso/lasso.py:480]: this is an implementation detail, so it is your call to decide what to do with. I think it is very helpful to have tests that specifically tests the transformation of messages to lasso representation and back from lasso representation to lite-llm. this is mostly where we can find edge cases . I suggest split every "case" (text message, text message with tool definitions, tool use, tool call etc) to a separate standalone pair of functions (to lasso-from lasso) which is easy to test. 
That's just a suggestion, feel free to so what ever you want with it

**Comment by vladpolevoi** [comment]
Closed due to moving to this https://github.com/BerriAI/litellm/pull/27648
